src/applications/personalization/appointments/actions/index.js

```diff
@@ -1,27 +1,212 @@
+      } else {
+        facilitiesResponse = await apiRequest(
+          `${environment.API_URL}/v1/facilities/va?ids=${facilityIDs.join(
+            ',',
+          )}`,
+        );
+      }
+
+      facilitiesLookup = facilitiesResponse?.data?.reduce(
+        (lookup, facility) => {
+          const facilityID = replace(facility?.id, 'vha_', '');
+          // eslint-disable-next-line no-param-reassign
+          lookup[facilityID] = facility;
+          return lookup;
+        },
+        {},
+      );
+    } catch (error) {
+      dispatch({
+        type: FETCH_CONFIRMED_FUTURE_APPOINTMENTS_FAILED,
       });
+    }
+
+    const formattedVAAppointments = vaAppointments.reduce(
+      (accumulator, appointment) => {
+        const startDate = moment(appointment?.attributes?.startDate);
+        const now = moment();
+        const appointmentStatus =
+          appointment?.attributes?.vdsAppointments?.[0]?.currentStatus;
+
+        // Past appointments should be filtered out already on the api side, this is a safe guard.
+        if (startDate.isBefore(now)) {
+          return accumulator;
+        }
+
+        // Hide cancelled or no-show appointments
+        if (FUTURE_APPOINTMENTS_HIDDEN_SET.has(appointmentStatus)) {
+          return accumulator;
+        }
+
+        // Derive the facility.
+        const facility =
+          facilitiesLookup[getStagingID(appointment?.attributes?.facilityId)];
+
+        accumulator.push({
+          additionalInfo: getAdditionalInfo(appointment),
+          facility,
+          id: appointment?.id,
+          isVideo: !isEmpty(appointment?.attributes?.vvsAppointments),
+          providerName: facility?.attributes?.name,
+          startsAt: startDate.toISOString(),
+          type: 'va',
+          timeZone: getVATimeZone(
+            getStagingID(appointment?.attributes?.facilityId),
+          ),
+        });
+
+        return accumulator;
+      },
+      [],
+    );
+
+    const formattedCCAppointments = ccAppointments.reduce(
+      (accumulator, appointment) => {
+        const startDate = moment(appointment?.attributes?.appointmentTime);
+        const now = moment();
+
+        // Escape early if the appointment is in the past.
+        if (startDate.isBefore(now)) {
+          return accumulator;
+        }
+
+        accumulator.push({
+          additionalInfo: getAdditionalInfo(appointment),
+          id: appointment?.id,
+          isVideo: false,
+          providerName: appointment?.attributes?.providerPractice,
+          startsAt: startDate.toISOString(),
+          timeZone: getCCTimeZone(appointment),
+          type: 'cc',
+        });
+
+        return accumulator;
+      },
+      [],
+    );
+
+    const allAppointments = [
+      ...formattedCCAppointments,
+      ...formattedVAAppointments,
+    ];
+
+    // Sort the appointments by which is soonest.
+    const sortedAppointments = allAppointments.sort((a, b) => {
+      return moment(a.startsAt).diff(b.startsAt);
```

Comment:
oh I should do something like this the sorting functions I'm using for Claims and Appeals